src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -5898,6 +5898,13 @@ void ThreadSuspend::RestartEE(BOOL bFinishedGC, BOOL SuspendSucceded)
 
     FireEtwGCRestartEEBegin_V1(GetClrInstanceId());
 
+#if defined(TARGET_ARM) || defined(TARGET_ARM64)
+    // Flush the store buffers on all CPUs, to ensure that they all see changes made
+    // by the GC threads. This only matters on weak memory ordered processors as 
+    // the strong memory ordered processors wouldn't have reordered the relevant reads.
+    ::FlushProcessWriteBuffers();
```

Comment:
Assume there are perf implications due to this? Do we know what the impact is?

Reply:
I haven't measured...I was gonna ask if you still have that arm64 machine I could borrow. but I don't expect that I can actually measure the perf cost of this as a GC cost would make this negligible.

Reply:
I dont have the arm64 machine still, but I can reserve one again if it would be helpful.